libusb/os/haiku_usb_raw.cpp

```diff
@@ -209,6 +209,15 @@ haiku_clock_gettime(int clkid, struct timespec *tp)
 	return LIBUSB_ERROR_INVALID_PARAM;
 }
 
+static int haiku_get_device_driver(struct libusb_device *device, char *driver,
+																 int size)
+{
+	if (size > 0 && device) {
+		driver[0] = '\0';
+	}
+	return 1;
+}
+
```

Comment:
Maybe it is possible to have a `default_get_device_driver() ` to not have the define the same function in 6 different backends. Only the windows backend uses a different implementation.

Reply:
I thought about it, but where would I put that function?

Reply:
Maybe in `core.c` since it is common code?

I am not sure the function should return an empty string on Unix systems. Maybe returning `LIBUSB_ERROR_NOT_SUPPORTED` is better.
In fact I don't know why this new API is needed. Is it mandatory for hotplug?
Maybe you should split the patch and move the addition of `libusb_get_driver()` in a different Pull Request.

Reply:
The addition to the API is important for us, not hotplugging.

**Why?**
Because some driver result in blue screens and that is something that has to be prevented. So this API is usefull for all windows users that have to make sure that a sane driver is used as backend. I thought I added that to the commit message but I must have forgotten it.

**API**
Ok that can be done.